include/openthread/platform/radio.h

```diff
@@ -621,6 +622,32 @@ bool otPlatRadioGetPromiscuous(otInstance *aInstance);
  */
 void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable);
 
+/**
+ * Indicates whether radio should stay in Receive or Sleep state during idle periods.
+ *
+ * Radio driver implementations supporting OT_RADIO_CAPS_RX_ON_WHEN_IDLE capability should consider idle periods the
+ * following situations:
+ *
+ * - Finalization of a regular frame reception task, provided that:
+ *   - The frame is received without errors and passes the filtering and it's not an spurious ACK.
+ *   - ACK is not requested or transmission of ACK is not possible due to internal conditions.
+ * - Finalization of a regular frame transmission task or transmission of an ACK frame, when ACK is not requested in
+ *   the transmitted frame.
+ * - Finalization of the reception operation of a requested ACK due to:
+ *   - ACK timeout expiration.
+ *   - Reception of an invalid ACK or not an ACK frame.
+ *   - Reception of the proper ACK, unless the transmitted frame was a Data Request Command and the frame pending bit
+ *     on the received ACK is set to true.
+ * - Finalization of a stand alone CCA task.
+ * - Finalization of a CCA operation with busy result during CSMA/CA procedure.
+ * - Finalization of an Energy Detection task.
```

Comment:
Thanks @edmont for adding this. It makes it clear and make sense 👍. 
I missed that we can make "receive" a one-shot request as well (i.e. we can ask radio to rx a frame and then quickly go to sleep).